### PR TITLE
Check length of subpacket when parsing signature

### PR DESCRIPTION
**Pull request by marinthiercelin**

### CODE REVIEW

File: openpgp/packet/signature.go [hunk 1]
```diff
@@ -1039,7 +1055,7 @@ func (sig *Signature) AddMetadataToHashSuffix() {
 	n := sig.HashSuffix[len(sig.HashSuffix)-8:]
 	l := uint64(
 		uint64(n[0])<<56 | uint64(n[1])<<48 | uint64(n[2])<<40 | uint64(n[3])<<32 |
-		uint64(n[4])<<24 | uint64(n[5])<<16 | uint64(n[6])<<8 | uint64(n[7]))
+			uint64(n[4])<<24 | uint64(n[5])<<16 | uint64(n[6])<<8 | uint64(n[7]))
```
Comment (twiss) [inline openpgp/packet/signature.go:1058]: Could you revert this (and some other) unrelated whitespace changes?

Comment (marinthiercelin) [reply openpgp/packet/signature.go:1058]: Those were changed automatically by my extension most likely using gofmt.
If I revert it, I'll have to manually do that every time I touch the file.
Is that really a problem ?

Comment (twiss) [reply openpgp/packet/signature.go:1058]: OK, maybe just make a separate PR with those whitespace changes (maybe for the entire project?) and then you can rebase this one on top of that

**Review by twiss — APPROVED** [review]
